lib/kube/proxy/forwarder.go

```diff
@@ -1553,10 +1601,10 @@ func setupImpersonationHeaders(log logrus.FieldLogger, ctx authContext, headers
 		switch header {
 		case ImpersonateUserHeader:
 			if impersonateUser != "" {
-				return trace.AccessDenied("%v, user already specified to %q", ImpersonationRequestDeniedMessage, impersonateUser)
+				return impersonateUser, impersonateGroups, trace.AccessDenied("%v, user already specified to %q", ImpersonationRequestDeniedMessage, impersonateUser)
```

Comment:
I noticed this looks different from v12. Since we are changing it, could we make the branches match more closely?